native/cocos/application/CocosApplication.cpp

```diff
@@ -52,8 +52,16 @@ int CocosApplication::init() {
     if (_engine->init()) {
         return -1;
     }
-    auto callback = std::bind(&CocosApplication::handleAppEvent, this, std::placeholders::_1); // NOLINT(modernize-avoid-bind)
-    _engine->addEventCallback(OSEventType::APP_OSEVENT, callback);
+
+    _engine->on(BaseEngine::ON_RESUME, [this]() {
+        this->onResume();
+    });
+    _engine->on(BaseEngine::ON_PAUSE, [this]() {
+        this->onPause();
+    });
+    _engine->on(BaseEngine::ON_CLOSE, [this]() {
+        this->onClose();
```

Comment:
Do we need to `off` the subscribing somewhere?

Reply:
Thanks for pointing out the mistake, I forgot to think about it. This also considers the restart part.